### PR TITLE
JNG-4696 Multiple instances of entity on one creation

### PR DESCRIPTION
**Pull request by bencegelei**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4696" title="JNG-4696" target="_blank">BugJNG-4696</a>  Multiple entity instances
  </td></table>
  <br />
 

**Issue**: during creation request handled by Dispatcher, there are some validations which contains range dao calls that - in order to query up-to-date content - creates/updates objects in the database. However these changes are not rolled back because they are not wrapped in an always rollback behavior (like during simple range calls).

**Fix**: validating and processing of parameters for creation process is contained in a transaction that is rolled back after operation.

### CODE REVIEW

**Review by robertcsakany — APPROVED** [review]

File: judo-runtime-core-dispatcher/src/main/java/hu/blackbelt/judo/runtime/core/dispatcher/DefaultDispatcher.java [hunk 1]
```diff
@@ -741,6 +743,24 @@ public Map<String, Object> callOperation(final String operationFullyQualifiedNam
...
+    private void validateAndConvertParameters(Map<String, Object> exchange, EOperation operation) {
+        final List<ValidationResult> ValidationResults = new ArrayList<>();
+        final List<EParameter> parameters = operation.getEParameters().stream()
+                                                     .filter(parameter -> (parameter.getEType() instanceof EClass))
+                                                     .collect(Collectors.toList());
+
+        parameters.forEach(parameter -> {
+            final EClass transferObjectType = (EClass) parameter.getEType();
+            final Map<String, Object> validationContext = new TreeMap<>();
+            validationContext.put(LOCATION_KEY, parameters.size() != 1 || parameter.isMany() ? parameter.getName() : "");
```
Comment (noherczeg) [inline judo-runtime-core-dispatcher/src/main/java/hu/blackbelt/judo/runtime/core/dispatcher/DefaultDispatcher.java:755]: Empty string is valid why exactly?

File: judo-runtime-core-dispatcher/src/main/java/hu/blackbelt/judo/runtime/core/dispatcher/behaviours/AlwaysRollbackTransactionalBehaviourCall.java [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
 @Slf4j
 public abstract class AlwaysRollbackTransactionalBehaviourCall<ID> implements BehaviourCall<ID> {
-    private static final String ROLLBACK_KEY = "ROLLBACK";
+    public static final String ROLLBACK_KEY = "ROLLBACK";
```
Comment (noherczeg) [inline judo-runtime-core-dispatcher/src/main/java/hu/blackbelt/judo/runtime/core/dispatcher/behaviours/AlwaysRollbackTransactionalBehaviourCall.java:36]: Can this be `protected` instead?

Comment (bencegelei) [reply judo-runtime-core-dispatcher/src/main/java/hu/blackbelt/judo/runtime/core/dispatcher/behaviours/AlwaysRollbackTransactionalBehaviourCall.java:36]: DefaultDispatcher uses this static field which would not have access to it if it were protected.

File: judo-runtime-core-dispatcher/src/main/java/hu/blackbelt/judo/runtime/core/dispatcher/DefaultDispatcher.java (hunk 1 above)
Comment (bencegelei) [reply judo-runtime-core-dispatcher/src/main/java/hu/blackbelt/judo/runtime/core/dispatcher/DefaultDispatcher.java:755]: I don't know tbh :D this part was only extracted as a method without any functional modification.

**Review by noherczeg — APPROVED** [review]

**Pull request merged**